src/components/ebay-snackbar-dialog/index.marko

```diff
@@ -0,0 +1,105 @@
+import processHtmlAttributes from "../../common/html-attributes"
+
+static var ignoredAttributes = [
+    "open"
+];
+
+class {
+    clearAllTimeouts() {
+        while(this.timeouts.length) {
+            clearTimeout(this.timeouts.pop());
+        }
+    }
+
+    onInput(input) {
+        this.clearAllTimeouts();
+        this.state = {open: input.open || this.state.open || false}
+        if (input.open !== false) {
+            this.timeouts.push(setTimeout(() => {
+                this.emit("close");
+                this.state.open = false;
+            }, input.timeout || this.DEFAULT_TIMEOUT_LENGTH));
+        }
+    }
+    onDestroy() {
+        this.clearAllTimeouts();
+    }
+
+    onCreate() {
+        this.DEFAULT_TIMEOUT_LENGTH = 6000;
+        this.eventSet = new Set();
+        this.state = {open: false};
+        this.timeouts = [];
```

Comment:
I'm not sure why is this array needed like this?

IMO, there should be 1 timeout to close the snackbar. 
Once you leave it should start to trigger the timeout. 
If you enter the snackbar, it clears the timeout. 
Why do we have multiple ones?

Reply:
So what was happening is in some cases multiple timeouts were created before they were cleared. Certain combinations of mouseLeave and Blur events were happening so that a couple timeouts were created before the first was cleared, and then the snackbar was timing out and closing too early later on. Having the array ensures that the timeouts are always cleared together.

Reply:
I did this change locally and it seems to be working (feel free to use it and let me know if there is a problem)
Only change I would do is remove the `clearAllTimeouts` and just do `clearTimeout(this.timeout)`
```
diff --git a/src/components/ebay-snackbar-dialog/index.marko b/src/components/ebay-snackbar-dialog/index.marko
index 6521528f..62bd96d3 100644
--- a/src/components/ebay-snackbar-dialog/index.marko
+++ b/src/components/ebay-snackbar-dialog/index.marko
@@ -6,19 +6,17 @@ static var ignoredAttributes = [
 
 class {
     clearAllTimeouts() {
-        while(this.timeouts.length) {
-            clearTimeout(this.timeouts.pop());
-        }
+        clearTimeout(this.timeout);
     }
 
     onInput(input) {
         this.clearAllTimeouts();
         this.state = {open: input.open || this.state.open || false}
         if (input.open !== false) {
-            this.timeouts.push(setTimeout(() => {
+            this.timeout = setTimeout(() => {
                 this.emit("close");
                 this.state.open = false;
-            }, input.timeout || this.DEFAULT_TIMEOUT_LENGTH));
+            }, input.timeout || this.DEFAULT_TIMEOUT_LENGTH);
         }
     }
     onDestroy() {
@@ -29,7 +27,6 @@ class {
         this.DEFAULT_TIMEOUT_LENGTH = 6000;
         this.eventSet = new Set();
         this.state = {open: false};
-        this.timeouts = [];
     }
 
     handleAction() {
@@ -47,10 +44,11 @@ class {
     handleBlur() {
         this.eventSet.delete("focus");
         if (this.eventSet.size === 0) {
-            this.timeouts.push(setTimeout(() => {
+            this.clearAllTimeouts();
+            this.timeout = setTimeout(() => {
                 this.emit("close");
                 this.state.open = false;
-            }, this.DEFAULT_TIMEOUT_LENGTH));
+            }, this.DEFAULT_TIMEOUT_LENGTH);
         }
     }
 
@@ -62,10 +60,11 @@ class {
     handleMouseLeave() {
         this.eventSet.delete("mouseOver");
         if (this.eventSet.size === 0 && this.state.open === true) {
-            this.timeouts.push(setTimeout(() => {
+            this.clearAllTimeouts();
+            this.timeout = setTimeout(() => {
                 this.emit("close");
                 this.state.open = false;
-            }, this.DEFAULT_TIMEOUT_LENGTH));
+            }, this.DEFAULT_TIMEOUT_LENGTH);
         }
     }
 ```